x-pack/examples/testing_embedded_lens/public/app.tsx

```diff
@@ -0,0 +1,679 @@
+  const initialAttributes = useMemo(() => JSON.stringify(charts[0].attributes, null, 2), []);
+
+  const [currentSO, saveSO] = useState(initialAttributes);
+  const switchChartPreset = useCallback(
+    (newIndex) => {
+      const newChart = charts[newIndex];
+      saveSO(JSON.stringify(newChart.attributes, null, 2));
+    },
+    [charts]
+  );
+
+  const currentAttributes = useMemo(() => JSON.parse(currentSO), [currentSO]);
+
+  const isDisabled = !currentAttributes;
+  const isColorDisabled = isDisabled || !isXYChart(currentAttributes);
+
+  return (
+    <KibanaContextProvider services={{ uiSettings: props.core.uiSettings }}>
+      <EuiPageTemplate fullHeight template="empty">
+        <EuiFlexGroup
+          className="eui-fullHeight"
+          gutterSize="none"
+          direction="column"
+          responsive={false}
+        >
+          <EuiFlexItem className="eui-fullHeight">
+            <EuiFlexGroup className="eui-fullHeight" gutterSize="l">
+              <EuiFlexItem grow={3}>
+                <EuiPanel hasShadow={false}>
+                  <p>
+                    This app embeds a Lens visualization by specifying the configuration. Data
+                    fetching and rendering is completely managed by Lens itself.
+                  </p>
+                  <p>
+                    The editor on the right hand side make it possible to paste a Lens attributes
+                    configuration, and have it rendered. Presets are available to have a starting
+                    configuration, and new presets can be saved as well (not persisted).
+                  </p>
+                  <p>
+                    The Open with Lens button will take the current configuration and navigate to a
+                    prefilled editor.
+                  </p>
+                  <EuiSpacer />
+                  <EuiFlexGroup wrap>
+                    <EuiFlexItem grow={false}>
+                      <EuiButton
+                        data-test-subj="lns-example-change-color"
+                        isLoading={isLoading}
+                        onClick={() => {
+                          // eslint-disable-next-line no-bitwise
+                          const newColor = '#' + ((Math.random() * 0xffffff) << 0).toString(16);
+                          saveSO(
+                            JSON.stringify(
+                              getLensAttributes(
+                                props.defaultDataView,
+                                fields,
+                                currentAttributes.state.visualization.preferredSeriesType,
+                                newColor
+                              ),
+                              null,
+                              2
+                            )
+                          );
+                        }}
+                        isDisabled={isColorDisabled}
+                      >
+                        Change color
```

Comment:
I noticed that you use this button to indicate when there's a loading state. I think it's enough for this demo, but it makes the whole panel jump down because of the button width changes. What if you change this copy to:
```suggestion
                        {isLoading ? 'Loading' : 'Change color'}
```
then the button stays the same width and it stays in place. 

Uploading jump.mp4…